test/core/libp2p/transport/tcp_edge_cases.cpp

```diff
@@ -139,33 +138,29 @@ TEST(TCP, SingleListenerCanAcceptManyClients) {
       boost::asio::io_context context;
 
       auto transport = std::make_unique<TransportImpl>(context);
-      for (int i = 0; i < kRetries; i++) {
-        srand(i);
-
-        EXPECT_OUTCOME_TRUE(conn, transport->dial(ma));
-
-        auto buf = std::make_shared<Buffer>(kSize, 0);
-        std::generate(buf->begin(), buf->end(), []() {
-          return rand();  // NOLINT
-        });
-
-        auto rdbuf = std::make_shared<Buffer>(kSize, 1);
-        conn->asyncWrite(
-            boost::asio::buffer(buf->toVector(), kSize),
-            [buf, kSize, conn, rdbuf](auto &&ec, size_t write) {
-              ASSERT_FALSE(ec);
-              ASSERT_EQ(kSize, write);
-
-              conn->asyncRead(
-                  boost::asio::buffer(rdbuf->toVector(), kSize), kSize,
-                  [buf, kSize, write, rdbuf](auto &&ec, size_t read) {
-                    ASSERT_FALSE(ec);
-                    ASSERT_EQ(read, write);
-                    ASSERT_EQ(read, kSize);
-                    ASSERT_EQ(*rdbuf, *buf);
-                  });
-            });
-      }
+      EXPECT_OUTCOME_TRUE(conn, transport->dial(ma));
+
+      auto buf = std::make_shared<Buffer>(kSize, 0);
+      std::generate(buf->begin(), buf->end(), []() {
+        return rand();  // NOLINT
+      });
+
+      auto rdbuf = std::make_shared<Buffer>(kSize, 1);
+      conn->asyncWrite(
+          boost::asio::buffer(buf->toVector(), kSize),
+          [buf, kSize, conn, rdbuf](auto &&ec, size_t write) {
+            EXPECT_ERRCODE_SUCCESS(ec);
+            ASSERT_EQ(kSize, write);
+
+            conn->asyncRead(boost::asio::buffer(rdbuf->toVector(), kSize),
+                            kSize,
+                            [buf, kSize, write, rdbuf](auto &&ec, size_t read) {
+                              EXPECT_ERRCODE_SUCCESS(ec);
+                              ASSERT_EQ(read, write);
+                              ASSERT_EQ(read, kSize);
+                              ASSERT_EQ(*rdbuf, *buf);
+                            });
+          });
```

Comment:
asyncWrite was executed in a loop, it made `conn` (socket) go out of scope, therefore some clients failed, which made test fail.

I removed forloop, therefore extending "scope" of a socket, so test can be considered as "fixed"